charts/infra-server/templates/server/configmap.yaml

```diff
@@ -37,8 +37,13 @@ data:
 {{- $config := unset $config "secrets" }}
 
 {{- if .Values.config.admin.enabled }}
-{{- $_ := set $config "users" (list (dict "name" "admin@local" "password" "file:/var/run/secrets/infrahq.com/initial-admin-secret/password")) }}
-{{- $_ := set $config "grants" (list (dict "user" "admin@local" "role" "admin" "resource" "infra")) }}
+{{- $adminUser := dict "name" "admin@local" }}
```

Comment:
Reading this, I was wondering if it be simpler to avoid abstracting away the `users:` field here and expose it directly as a key in the values file? (vs new helm-level abstractions like `admin.accessKeySecret`)?

E.g.

```
server:
  config:
    users:
      name: "admin@myco.net"
      accessKey: <path to access key to use>
```

Then, if `users` is blank, we could generate an initial admin user, access key and password for them that are accessible in Kubernetes secrets for easy deployment.

Reply:
We talked about this in Zoom but putting it here for posterity.

Adding `users` here makes it much harder to configure a basic working configuration. The user will be responsible for creating the bootstrap user, or if they neglect to do that, the helm chart will need to do it for them. This means looking through the `users` list and finding if an Infra admin exists. 

Furthermore loading values for passwords and access keys into this format is not trivial since users will need to understand the Infra `env:` and `file:` format as well as requiring additional Helm values for `env`, `envFrom` or `volumes` and `volumeMounts` in order to set these to non-plaintext values.

For advanced users, this can certainly be an option but the base case should be simple and easy to configure while setting up the user for success.